javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,158 @@
+let savedName = " ";
+const toDoList = [];
+
+/**
+ * function to get Reply from voice assistant
+ * using input as command
+ * @param {*} command
+ */
+function getReply(command) {
+  if (typeof command != "string" || typeof command == undefined) {
+    return "Invalid input";
+  }
+  const input = command.toLowerCase();
+  let activity;
+
+  if (input.includes("hello my name is")) {
+    let token = input.split(" ");
+    const username = token[token.length - 1];
+    if (savedName === " ") {
+      savedName = username;
+      return `Nice to meet you ${username}`;
+    } else if (savedName === username) {
+      return `Nice to meet you again ${username}`;
+    }
+  } else if (input.includes("what is my name")) {
+    if (savedName === " ") {
+      return `Sorry I dont know your name`;
+    } else {
+      return `Your name is ${savedName}`;
+    }
+  } else if (input.includes("add") && input.includes("todo")) {
+    activity = input.match(/add (.*?) to my todo/)[1];
```

Comment:
Regexes can be very powerful, also pretty hard to debug.
Luckily, there are some [nice tools](https://regex-vis.com/) out there.